codes/data/__init__.py

```diff
@@ -1,55 +1,66 @@
-'''create dataset and dataloader'''
+"""Create dataset and dataloader"""
 import logging
+
 import torch.utils.data
 
-def create_dataloader(dataset, dataset_opt):
-    '''create dataloader '''
-    phase = dataset_opt['phase']
-    if phase == 'train':
-        return torch.utils.data.DataLoader(
-            dataset,
-            batch_size=dataset_opt['batch_size'],
-            shuffle=dataset_opt['use_shuffle'],
-            num_workers=dataset_opt['n_workers'],
-            drop_last=True,
-            pin_memory=True)
-    else:
-        return torch.utils.data.DataLoader(
-            dataset, batch_size=1, shuffle=False, num_workers=1, pin_memory=True)
 
+def create_dataloader(dataset: torch.utils.data.Dataset, dataset_opt: dict) -> torch.utils.data.DataLoader:
+    """
+    Create Dataloader.
+    :param dataset: Dataset to use
+    :param dataset_opt: Dataset configuration from opt file
+    """
+    batch_size = 1
+    shuffle = False
+    num_workers = 0
```

Comment:
Update to 1

Reply:
It's set to 0 for a specific reason. Since 1 will use a seperate process, it will be an unnecessary slowdown. Since this will be set to 1 if its for val or test purposes, its going to be going 1 at a time anyhow, so it being called in the main process can only make it faster.

Reply:
No, the main process in the pytorch dataloader should only be the driver process. Other processes are threaded separatedly so they don't crash the driver process if they fail. It doesn't cause slowdowns, it paralelizes the process using more hardware threads